src/driver/mongodb.rs

```diff
@@ -22,7 +22,7 @@ impl Driver for MongoDriver {
   fn connect(uri: &str) -> Result<Self, Error> {
     let config = try!(connstring::parse(uri));
 
-    if let Some(db_name) = config.clone().database {
+    if let Some(db_name) = config.database.clone() {
```

Comment:
I was talking about using `&` to coerce String into `&str`, but if it is too much work, just leave it as is.